src/main/java/org/isf/menu/service/MenuIoOperations.java

```diff
@@ -313,19 +299,16 @@ private GroupMenu insertGroupMenu(UserGroup aGroup, UserMenuItem item) throws OH
 
 	/**
 	 * Deletes a {@link UserGroup}
-	 * 
 	 * @param aGroup - the {@link UserGroup} to delete
 	 * @throws OHServiceException When failed to delete group
 	 */
 	public void deleteGroup(UserGroup aGroup) throws OHServiceException {
-		groupMenuRepository.deleteWhereUserGroup(aGroup.getCode());
-		groupPermissionIoOperationRepository.deleteAllByUserGroup_Code(aGroup.getCode());
-		groupRepository.delete(aGroup);
+		aGroup.setDeleted(true);
```

Comment:
I am going to stop asking about whether checks for deletion should be made in update/delete methods.   But a consistent answer is needed.